### PR TITLE
Week 3 - Part 1

### PR DESCRIPTION
**Pull request by callieapan**

### CODE REVIEW

**Comment by Atcold** [comment]
Is it completed? Am I supposed to merge?
Can you add the summary to the PR first post?

File: docs/chapters/03-1.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
+---
+title: A name
```
Comment (Atcold) [inline docs/chapters/03-1.md]: This is **not** a title!

File: docs/chapters/03-1.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+authors:
```
Comment (Atcold) [inline docs/chapters/03-1.md]: There are no authors!!!

File: docs/chapters/03-1.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+
+
+​									   (a)									              (b)
+
+<div align="center">Fig. 3 Visualization of ReLU operator<div>
+
+
+
+
```
Comment (Atcold) [inline docs/chapters/03-1.md]: What are all these empty lines?

File: docs/chapters/03-1.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+w \leftarrow w - \eta\frac{\partial H}{\partial u}\frac{\partial H}{\partial u}^T\frac{\partial C}{\partial w}^T
+$$
+
+These formula is applied in a matrix form. Note that the dimension of right terms should be consistent. The dimension of $u$,$w$,$\frac{\partial H}{\partial u}^T$,$\frac{\partial C}{\partial w}^T$ are $[N_u \times 1]$,$[N_w \times 1]$,$[N_u \times N_w]$,$[N_w \times 1]$, respectively. Therefore, the dimension of backprop formula is consistent.
+
+<img src="./03-1/PT.png" alt="Network" style="zoom:35%;" />
+
+<div align="center">Fig. 5 General Form of Parameter Transformations<div>
+
+#### A Simple Parameter Transformation: Weight Sharing
```
Comment (Atcold) [inline docs/chapters/03-1.md]: Missing timestamp.

File: docs/chapters/03-1.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+A Weight Sharing Transformation means $H(u)$ just replicates one component of $u$ into multiple components of $w$. $H(u)$ is like a **Y** branch to copy $u_1$ to $w_1$, $w_2$. This can be expressed as,
+$$
+w_1 = w_2 = u_1, w_3 = w_4 = u_2
+$$
+We force shared parameters to be equal, so the gradient w.r.t. shared parameters will be summed in the backprop. So for example the gradient of the cost function $C(y, \bar y)$ with respect to $u_1$ will be the sum of the gradient of the cost function $C(y, \bar y)$ with respect to $w_1$ and the gradient of the cost function $C(y, \bar y)$ with respect to $w_2$
+
+#### Hypernetwork
```
Comment (Atcold) [inline docs/chapters/03-1.md]: Missing timestamp.

File: docs/chapters/03-1.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+ Hypernetwork is a network where the weights of one network is the output of another network. Figure 6 shows a framework of "hypernetwork". Here the function $H$ as a network with the parameter vector $u$ and the input $x$. As a result, the weights of $G(x,w)$ are dynamiclly configured by network $H(x,u)$. This idea is old but very powerful even at present.
+
+<img src="./03-1/hypernetwork.png" alt="Network" style="zoom:35%;" />
+
+<div align="center">Fig. 6 "Hypernetwork"<div>
+
+
+
+
+#### Motif Detection in Sequential Data
```
Comment (Atcold) [inline docs/chapters/03-1.md]: Missing timestamp.

File: docs/chapters/03-1.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+<img src="./03-1/motif.png" alt="Network" style="zoom:30%;" />
+
+<div align="center">Fig. 7 Motif Detection for Sequential Data<div>
+
+
+
+
+In this example we have 5 of those functions. As a result of this solution, we sum up five gradients and backpropagate to the parameter $w$. We do not want the implicit accumulation of these gradients, so we need to use zero_grad() in PyTorch to initialize the gradient.
+
+#### Motif Detection in Images
```
Comment (Atcold) [inline docs/chapters/03-1.md]: Missing timestamp.

File: docs/chapters/03-1.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+### Higher dimensional convolution
+
+For two dimensional inputs such as images, we make use of the two dimensional version of convolution:
+
+$$y_{ij} = \sum_{kl} w_{kl} x_{i+k, j+l}$$
+
+This definition can be extended beyond two dimensions to three, four dimensions similarly. Here $w$ is called the *convolution kernel*
+
+
+### Regular twists that can be made with the convolutional operator in DCNNs
```
Comment (Atcold) [inline docs/chapters/03-1.md]: Missing timestamps everywhere.

**Comment by callieapan** [comment]
Morning, 
I made the changes you mentioned above. 

As for summary, do you mean a summary of the entire note? If so it is below

For the first part of the lecture we will review a visualization of the how the input data is transformed by each of the 6-layer neural network and the non linear (Relu) function activated after each hidden layer (there are 3). Next we begin to the topic of Convolution and Convolution Neural Networks (CNN). We review several types of parameter transformation, and how motifs, in sound or images, can be detects by weight sharing techniques. Furthermore, in CNN we introduce the idea of a kernel, a "window function" that we run across the data to detects motifs. We can pass the outputs from that window as it strides across the data into a pooling function (often times a max function) to detects features. We repeat this process and learn features in a hierarchical manner, and use those features to classify our input data. Such is the basic idea of a CNN. Lastly, the design of CNN and Deep Learning is inspired by how we humans learn about our world. It seems incredible that we can comprehend our world by breaking down complex physical world into compositions (objects, components, compounds), and the art of Deep Learning is to imitate that process with computers.

**Comment by Atcold** [comment]
We're expecting a 50±10 word summary. See spreadsheet "red flag".

Screenshot 2020-02-16 16 41 07

**Comment by callieapan** [comment]
okay I'll reduce wc

**Comment by Atcold** [comment]
Simon_Thorpe.PNG caption missing.

**Comment by Atcold** [comment]
There are two figures 1.

**Pull request merged**